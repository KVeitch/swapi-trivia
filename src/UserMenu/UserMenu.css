.user__box {
    align-items: center;
    background-color: rgba(67, 52, 45, 0.7);
    color: white;
    display: flex;
    font-family: 'Barlow Condensed', sans-serif;
    flex-direction: row;
    font-size: 1.5em;
    justify-content: space-evenly;
    height: 15vh;
    margin-left: 7%;
    text-shadow: 3px 4px 4px rgba(23, 6, 1, 0.94);
    width: 85%
}

.box__ends {
    font-size: 1.7em;
}

.sign-out__button {
    color: white;
    font-size: .7em;
    height: 30%;
    width: 10%;
    border-radius: 5%;
    box-shadow: 0 10px 6px -6px rgba(67, 52, 45, 0.7);
    background-color:rgb(223, 213, 207);
    margin-top: 15px;
    cursor: pointer;
}