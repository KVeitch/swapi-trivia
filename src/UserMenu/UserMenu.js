import React from 'react';
import { Link } from 'react-router-dom';
import './UserMenu.css';

const UserMenu = (props) => {
    return(
        <div className='user__box'>
            <p className='box__ends'>{props.user}</p>
            <p>Favorite Quote: "{props.userQuote}"</p>
            <p className='box__ends'>Ranking: {props.userRanking}</p>
            <Link to='/movies/'><button type='button' onClick={props.resetIsCurrentCharacterLoaded}>Back To Movies</button></Link> 
            <button onClick={props.userSignOut}>Sign Out</button>
        </div>
    )
}

export default UserMenu;