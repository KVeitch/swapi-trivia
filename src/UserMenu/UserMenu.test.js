import React from 'react';
import { shallow } from 'enzyme';
import UserMenu from './UserMenu';

describe('User Menu', () => {
    let wrapper;
    let mockResetIsCurrentCharacterLoaded = jest.fn()
    it('should match the snapshot with all the data passed in correctly', () => {
        wrapper =
            shallow(<UserMenu
                    user={''}
                    userQuote={''}
                    userRanking={'Nerf Herder'}
                    resetIsCurrentCharacterLoaded={mockResetIsCurrentCharacterLoaded}/>)
        expect(wrapper).toMatchSnapshot();
    });
});