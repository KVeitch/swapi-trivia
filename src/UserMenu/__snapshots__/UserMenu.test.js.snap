// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Menu should match the snapshot with all the data passed in correctly 1`] = `
<div
  className="footer"
>
  <div
    className="user__box"
  >
    <p
      className="box__ends"
    />
    <p>
      Favorite Quote: "
      "
    </p>
    <p
      className="box__ends"
    >
      Ranking: 
      Nerf Herder
    </p>
  </div>
  <div
    className="buttons"
  >
    <Link
      to="/movies/"
    >
      <button
        className="user__buttons"
        onClick={[MockFunction]}
      >
        Back To Movies
      </button>
    </Link>
    <Link
      to="/favorites/"
    >
      <button
        className="user__buttons"
      >
        Favorite Characters
      </button>
    </Link>
    <Link
      to="/"
    >
      <button
        className="user__buttons"
      >
        Sign Out
      </button>
    </Link>
  </div>
</div>
`;
